core/src/main/java/com/graphhopper/routing/util/NameSimilarityEdgeFilter.java

```diff
@@ -171,9 +171,17 @@ public final boolean accept(EdgeIteratorState iter) {
         return isJaroWinklerSimilar(pointHint, edgeName);
     }
 
-    private static BBox createBBox(NodeAccess na, EdgeIteratorState edgeState) {
-        return BBox.fromPoints(na.getLat(edgeState.getBaseNode()), na.getLon(edgeState.getBaseNode()),
-                na.getLat(edgeState.getAdjNode()), na.getLon(edgeState.getAdjNode()));
+    private static BBox createBBox(EdgeIteratorState edgeState) {
+        // we should include the entire geometry, see #2319
+        PointList geometry = edgeState.fetchWayGeometry(FetchMode.ALL);
+        BBox bbox = new BBox(180, -180, 90, -90);
+        for (int i = 0; i < geometry.getSize(); i++) {
+            bbox.minLon = Math.min(bbox.minLon, geometry.getLon(i));
+            bbox.maxLon = Math.max(bbox.maxLon, geometry.getLon(i));
+            bbox.minLat = Math.min(bbox.minLat, geometry.getLat(i));
+            bbox.maxLat = Math.max(bbox.maxLat, geometry.getLat(i));
+        }
```

Comment:
Finding the bounds of a point list is  maybe something we do elsewhere already and we could create a helper method for this otherwise. Unfortunately I could not find something like it. 

Btw. why has PointList `getSize()` and `size()`? And looking at point list I was not even sure if `size()` really returns the number of elements or rather the capacity?!

Reply:
Maybe we should store the bounds of each edge including the geometry for such use cases. There are two very similar todos here: 

https://github.com/graphhopper/graphhopper/blob/e806a5c55750555c1c8031752b6ceae601a33eb3/core/src/main/java/com/graphhopper/storage/GraphEdgeIdFinder.java#L226-L228

https://github.com/graphhopper/graphhopper/blob/0522c495ccad5f22fe19c6e858a40a7bd6f74eae/core/src/main/java/com/graphhopper/routing/weighting/custom/CustomWeightingHelper.java#L61-L63